### PR TITLE
[DOCS-7373] Add full-text search

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do? What is the motivation?

Adds information about Full-text search.

[DOCS-7373](https://datadoghq.atlassian.net/browse/DOCS-7373)

### Merge instructions

- [ ] Please merge after reviewing

### Additional notes

[DOCS-7373]: https://datadoghq.atlassian.net/browse/DOCS-7373?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/logs/explorer/search_syntax.md [hunk 1]
```diff
@@ -113,13 +115,41 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
...
 ```
 *NETWORK*
 ```
 
-However, this search term does not return logs that contain the string `NETWORK` if it is in a facet and not part of the log message.
+However, this search term does not return logs that contain the string `NETWORK` if it is in a log attribute and not part of the log message.
+
+### Full-text search across all log attributes
+
+<div class="alert alert-warning">The full-text search feature is in beta. It is only available in Log Management and works for monitors, dashboards, and notebooks. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:128]:
```suggestion
<div class="alert alert-warning">The full-text search feature is in beta. It is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```

File: content/en/logs/explorer/search_syntax.md [hunk 2]
```diff
@@ -113,13 +115,41 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
...
 ```
 *NETWORK*
 ```
 
-However, this search term does not return logs that contain the string `NETWORK` if it is in a facet and not part of the log message.
+However, this search term does not return logs that contain the string `NETWORK` if it is in a log attribute and not part of the log message.
+
+### Full-text search across all log attributes
+
+<div class="alert alert-warning">The full-text search feature is in beta. It is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:128]: > works in monitor, dashboard, and notebook queries.

@siddhingra can you confirm this is correct?

File: content/en/logs/explorer/search_syntax.md [hunk 3]
```diff
@@ -113,13 +115,43 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
...
+
+<div class="alert alert-warning">You can use full-text search for all logs ingested after MONTH DATE YEAR.</div>
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:130]: @siddhingra could you let me know the date the write path was turned on?

File: content/en/logs/explorer/search_syntax.md [hunk 4]
```diff
@@ -113,13 +115,43 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
 service:*mongo
 ```
 
-Wildcard searches can also be used to search in the plain text of a log that is not part of a facet. This query returns all the logs that contain the string `NETWORK`:
+Wildcard searches can also be used to search in the plain text of a log that is not part of a log attribute. For example, this query returns all the logs that contain the string `NETWORK`:
```
Comment (ckelner) [inline content/en/logs/explorer/search_syntax.md:118]:
```suggestion
Wildcard searches can also be used to search in the plain text of a log that is not part of a log attribute. For example, this query returns all the logs with content (message) that contain the string `NETWORK`:
```

File: content/en/logs/explorer/search_syntax.md [hunk 5]
```diff
@@ -113,13 +115,43 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
...
 ```
 *NETWORK*
 ```
 
-However, this search term does not return logs that contain the string `NETWORK` if it is in a facet and not part of the log message.
+However, this search term does not return logs that contain the string `NETWORK` if it is in a log attribute and not part of the log message.
+
+### Full-text search across all log attributes
+
+<div class="alert alert-warning">The full-text search feature is in beta. It is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```
Comment (ckelner) [inline content/en/logs/explorer/search_syntax.md:128]: I no longer believe this is beta. It was shipped to everyone as of March 25 2024 AFAIK / AFAICT: https://dd.slack.com/archives/C057S9CHMSB/p1711361111826889?thread_ts=1710492712.079139&cid=C057S9CHMSB

File: content/en/logs/explorer/search_syntax.md [hunk 6]
```diff
@@ -113,13 +115,43 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
...
+
+<div class="alert alert-warning">You can use full-text search for all logs ingested after MONTH DATE YEAR.</div>
+
+Use the syntax `*:search_term` to perform a free text search across all log attributes, including the log message.
+
+#### Single term example
```
Comment (ckelner) [inline content/en/logs/explorer/search_syntax.md:134]: Maybe let's add an example of the difference between `*:kelner` vs `*:*kelner*` ? See these two screenshots

Screenshot 2024-03-26 at 2 40 07 PM
vs
Screenshot 2024-03-26 at 2 40 29 PM

1,308 results vs 1,907,477 results

**Review by ckelner — COMMENTED** [review]
dropped some single comments outside the review, but highlighted the difference between `*:kelner` and `*:*kelner*` -- might be worth highlighting / calling out the use of wildcards within full-text search (whole-event search) as well?

File: content/en/logs/explorer/search_syntax.md (hunk 5 above)
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:128]:
```suggestion
<div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```

File: content/en/logs/explorer/search_syntax.md [hunk 7]
```diff
@@ -113,13 +115,43 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
...
 ```
 *NETWORK*
 ```
 
-However, this search term does not return logs that contain the string `NETWORK` if it is in a facet and not part of the log message.
+However, this search term does not return logs that contain the string `NETWORK` if it is in a log attribute and not part of the log message.
+
+### Full-text search across all log attributes
+
+<div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:128]: > works in monitor, dashboard, and notebook queries.

@siddhingra can you confirm this is correct?

File: content/en/logs/explorer/search_syntax.md (hunk 6 above)
Comment (maycmlee) [reply content/en/logs/explorer/search_syntax.md:133]: @ckelner Added an example below the single term example, is that what you are looking for?

**Review by ckelner — APPROVED** [review]
LGTM!

File: content/en/logs/explorer/search_syntax.md [hunk 8]
```diff
@@ -113,13 +114,50 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
...
+
+<div class="alert alert-warning">You can use full-text search for all logs ingested after MONTH DATE YEAR.</div>
+
+Use the syntax `*:search_term` to perform a free text search across all log attributes, including the log message.
+
+#### Single term example
+
+| Search syntax | Search type | Description                                           |
+| ------------- | ----------- | ----------------------------------------------------- |
+| `*:("hello")` | Full-text   | Searches all log attributes for the term `hello`.     |
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:137]:
```suggestion
| `*:hello` | Full-text   | Searches all log attributes for the term `hello`.     |
```

File: content/en/logs/explorer/search_syntax.md [hunk 9]
```diff
@@ -113,13 +114,50 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
...
+| `hello`       | Free text   | Searches only the log message for the term `hello`.   |
+
+#### Search term with wildcard example
+
+| Search syntax | Search type | Description                                                                                  |
+| ------------- | ----------- | -------------------------------------------------------------------------------------------- |
+| `*:("hello")` | Full-text   | Searches all log attributes for the exact string `hello`.                                    |
+| `*:("hello"*)`| Full-text   | Searches all log attributes for strings that starts with `hello`. For example, `hello_world`.|
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:145]:
```suggestion
| `*:hello` | Full-text   | Searches all log attributes for the exact string `hello`.                                    |
| `*:hello*`| Full-text   | Searches all log attributes for strings that starts with `hello`. For example, `hello_world`.|
```

File: content/en/logs/explorer/search_syntax.md [hunk 10]
```diff
@@ -113,13 +114,50 @@ Wildcard searches work within tags and attributes (faceted or not) with this syn
...
+
+#### Multiple terms with exact match example
+
+| Search syntax       | Search type | Description                                            |
+| ------------------- | ----------- |------------------------------------------------------- |
+| `*:("hello world")` | Full-text   |Searches all log attributes for the term `hello world`. |
+| `hello world`       | Free text   |Searches only the log message for the term `hello`.     |
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:152]:
```suggestion
| `*:hello world` | Full-text   |Searches all log attributes for the term `hello world`. |
| `hello world`       | Free text   |Searches only the log message for the term `hello`.     |
```

File: content/en/logs/explorer/search_syntax.md (hunk 6 above)
Comment (maycmlee) [reply content/en/logs/explorer/search_syntax.md:133]: DM'd this works 🎉

File: content/en/logs/explorer/search_syntax.md (hunk 7 above)
Comment (usman-dd) [reply content/en/logs/explorer/search_syntax.md:127]: yes, this is correct

File: content/en/logs/explorer/search_syntax.md (hunk 3 above)
Comment (usman-dd) [reply content/en/logs/explorer/search_syntax.md:129]: this feels way too prominent, maybe we don't need to call this out at all

**Comment by usman-dd** [comment]
Only one change needed before greenlighting this initial update --> "Full-text search" as its own top-level section, rather than "Full-text search across all log attributes" within the Wildcards section

File: content/en/logs/explorer/search_syntax.md (hunk 3 above)
Comment (maycmlee) [reply content/en/logs/explorer/search_syntax.md]: removed

File: content/en/logs/explorer/search_syntax.md (hunk 5 above)
Comment (maycmlee) [reply content/en/logs/explorer/search_syntax.md]: removed

File: content/en/logs/explorer/search_syntax.md [hunk 11]
```diff
@@ -39,6 +39,41 @@ To combine multiple terms into a complex query, you can use any of the following
...
+| Search syntax | Search type | Description                                                                                  |
+| ------------- | ----------- | -------------------------------------------------------------------------------------------- |
+| `*:hello` | Full-text   | Searches all log attributes for the exact string `hello`.                                    |
+| `*:hello*`| Full-text   | Searches all log attributes for strings that starts with `hello`. For example, `hello_world`.|
+
+#### Multiple terms with exact match example
+
+| Search syntax       | Search type | Description                                            |
+| ------------------- | ----------- |------------------------------------------------------- |
+| `*:"hello world"` | Full-text   |Searches all log attributes for the term `hello world`. |
```
Comment (alai97) [inline content/en/logs/explorer/search_syntax.md:66]:
```suggestion
| `*:"hello world"` | Full-text   | Searches all log attributes for the term `hello world`. |
```

File: content/en/logs/explorer/search_syntax.md [hunk 12]
```diff
@@ -39,6 +39,41 @@ To combine multiple terms into a complex query, you can use any of the following
...
+| `hello world`       | Free text   |Searches only the log message for the term `hello`.     |
```
Comment (alai97) [inline content/en/logs/explorer/search_syntax.md:67]:
```suggestion
| `hello world`       | Free text   | Searches only the log message for the term `hello`.     |
```

File: content/en/logs/explorer/search_syntax.md [hunk 13]
```diff
@@ -39,6 +39,41 @@ To combine multiple terms into a complex query, you can use any of the following
 | `OR`         | **Union**: either term is contained in the selected events                                             | authentication OR password   |
 | `-`          | **Exclusion**: the following term is NOT in the event (apply to each individual raw text search)                                                  | authentication AND -password |
 
+### Full-text search across all log attributes
+
+<div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```
Comment (alai97) [inline content/en/logs/explorer/search_syntax.md:44]:
```suggestion
<div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used to define index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```

**Review by alai97 — APPROVED** [review]
Looks good, some small suggestions!

File: content/en/logs/explorer/search_syntax.md [hunk 14]
```diff
@@ -39,6 +39,41 @@ To combine multiple terms into a complex query, you can use any of the following
 | `OR`         | **Union**: either term is contained in the selected events                                             | authentication OR password   |
 | `-`          | **Exclusion**: the following term is NOT in the event (apply to each individual raw text search)                                                  | authentication AND -password |
 
+### Full-text search across all log attributes
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:42]:
```suggestion
## Full-text search 
```

File: content/en/logs/explorer/search_syntax.md [hunk 15]
```diff
@@ -39,6 +39,41 @@ To combine multiple terms into a complex query, you can use any of the following
 | `OR`         | **Union**: either term is contained in the selected events                                             | authentication OR password   |
 | `-`          | **Exclusion**: the following term is NOT in the event (apply to each individual raw text search)                                                  | authentication AND -password |
 
+## Full-text search 
+
+<div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used to define index filters, archive filters, log pipeline filters, or in Live Tail. </div>
+
+Use the syntax `*:search_term` to perform a full-text search across all log attributes, including the log message.
+
+#### Single term example
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:48]:
```suggestion
### Single term example
```

File: content/en/logs/explorer/search_syntax.md [hunk 16]
```diff
@@ -39,6 +39,41 @@ To combine multiple terms into a complex query, you can use any of the following
...
+
+| Search syntax | Search type | Description                                           |
+| ------------- | ----------- | ----------------------------------------------------- |
+| `*:hello` | Full-text   | Searches all log attributes for the term `hello`.     |
+| `hello`       | Free text   | Searches only the log message for the term `hello`.   |
+
+#### Search term with wildcard example
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:55]:
```suggestion
### Search term with wildcard example
```

File: content/en/logs/explorer/search_syntax.md [hunk 17]
```diff
@@ -39,6 +39,41 @@ To combine multiple terms into a complex query, you can use any of the following
...
+
+| Search syntax | Search type | Description                                                                                  |
+| ------------- | ----------- | -------------------------------------------------------------------------------------------- |
+| `*:hello` | Full-text   | Searches all log attributes for the exact string `hello`.                                    |
+| `*:hello*`| Full-text   | Searches all log attributes for strings that starts with `hello`. For example, `hello_world`.|
+
+#### Multiple terms with exact match example
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:62]:
```suggestion
### Multiple terms with exact match example
```

File: content/en/logs/explorer/search_syntax.md [hunk 18]
```diff
@@ -39,6 +39,41 @@ To combine multiple terms into a complex query, you can use any of the following
...
+
+| Search syntax       | Search type | Description                                            |
+| ------------------- | ----------- |------------------------------------------------------- |
+| `*:"hello world"` | Full-text   | Searches all log attributes for the term `hello world`. |
+| `hello world`       | Free text   | Searches only the log message for the term `hello`.     |
+
+#### Multiple terms without exact match example
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:69]:
```suggestion
### Multiple terms without exact match example
```

File: content/en/logs/explorer/search_syntax.md [hunk 19]
```diff
@@ -39,6 +39,41 @@ To combine multiple terms into a complex query, you can use any of the following
...
+
+The full-text search syntax `*:hello world` is equivalent to `*:hello *:world`. It searches all log attributes for the terms `hello` and `world`.
+
+#### Multiple terms with a white space example
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:73]:
```suggestion
### Multiple terms with a white space example
```

**Review by usman-dd — APPROVED** [review]

**Pull request merged**